lib/python/rosie/svn_pre_commit.py

```diff
@@ -207,39 +134,64 @@ def run(self, repos, txn):
                 bad_changes.append(BadChange(status, path))
                 continue
 
-            # Can only add directories at levels above the suites
+            # At levels above the suites, can only add directories
             if len(names) < self.LEN_ID:
-                if status[0] != self.ST_ADD:
+                if status[0] != self.ST_ADDED:
                     bad_changes.append(BadChange(status, path))
                 continue
-            else:
-                is_meta_suite = "".join(names[0:self.LEN_ID]) == "ROSIE"
 
-            # A file at the branch level
+            # Cannot have a file at the branch level
             if len(names) == self.LEN_ID + 1 and tail is None:
                 bad_changes.append(BadChange(status, path))
                 continue
 
-            # No need to check non-trunk changes
-            if len(names) > self.LEN_ID and names[self.LEN_ID] != "trunk":
-                continue
-
             # New suite should have an info file
-            if status[0] == self.ST_ADD and len(names) == self.LEN_ID:
-                if (self.ST_ADD, path + "trunk/") not in changes:
+            if len(names) == self.LEN_ID and status == self.ST_ADDED:
+                if (self.ST_ADDED, path + "trunk/") not in changes:
                     bad_changes.append(
                         BadChange(status, path, BadChange.NO_TRUNK))
                     continue
                 path_trunk_info_file = path + self.TRUNK_INFO_FILE
-                if ((self.ST_ADD, path_trunk_info_file) not in changes and
-                        (self.ST_UPDATE, path_trunk_info_file) not in changes):
+                if ((self.ST_ADDED, path_trunk_info_file) not in changes and
+                        (self.ST_UPDATED,
+                         path_trunk_info_file) not in changes):
                     bad_changes.append(
                         BadChange(status, path, BadChange.NO_INFO))
                 continue
 
-            # The rest are trunk changes in a suite
-            path_head = "/".join(names[0:self.LEN_ID]) + "/"
+            sid = "".join(names[0:self.LEN_ID])
+            branch = names[self.LEN_ID] if len(names) > self.LEN_ID else None
+            path_head = "/".join(sid) + "/"
```

Comment:
Is this definitely equivalent to the original logic?

```python
# before
path_head = "/".join(names[0:self.LEN_ID]) + "/"
```

```python
# after
sid = "".join(names[0:self.LEN_ID])
path_head = "/".join(sid) + "/"
# expanded
path_head = "/".join("".join(names[0:self.LEN_ID])) + "/"
```

Reply:
Yes,
```python
"/".join("ab123")
```
is the same as
```python
"/".join(["a", "b", "1", "2", "3"])
```
if that's what you're asking